pkg/policy/mapstate.go

```diff
@@ -158,7 +170,15 @@ func (keys MapState) DenyPreferredInsert(newKey Key, newEntry MapStateEntry) {
 					}
 				}
 			}
-		} else {
+		case allCpy == newKey:
+			// If we adding a deny "all" entry, then we will remove all allow
+			// entries from the map state for that direction.
```

Comment:
We could also delete all deny entries in addition to the allow, as these more specific denies would be covered by the deny "all" entry.